### PR TITLE
Wait for geolocation and set result in main process

### PR DESCRIPTION
**Pull request by VladimirSlavik**
This makes geolocation completely separated from the UI.

Tested with both GUI and TUI.

Port of #4328 and further changes added.

### CODE REVIEW

File: pyanaconda/startup_utils.py [hunk 1]
```diff
@@ -646,3 +648,96 @@ def check_if_geolocation_should_be_used(opts):
...
+        log.info("Geoloc: skipping locale because already set")
+        return
+
+    territory = geoloc_result.territory
+    locales = get_territory_locales(territory)
+    try:
+        locale = next(l for l in locales if locale_has_translation(l))
+    except StopIteration:
+        log.info("Geoloc: detected languages are not translated, falling back to %s", DEFAULT_LANG)
+        locale = DEFAULT_LANG
```
Comment (poncovka) [inline pyanaconda/startup_utils.py:738]: Maybe, we could just return in this case. The locale is probably set to `DEFAULT_LANG` anyway in most cases.

**Review by poncovka — APPROVED** [review]
Thanks!

**Comment by VladimirSlavik** [comment]
Needs retesting after the early return added above, but not that much.

**Pull request merged**